File: enode/src/main/java/org/enodeframework/domain/IAggregateRepositoryProvider.java
package org.enodeframework.domain;

/**
 * Represents a provider to provide the aggregate repository.
 */
public interface IAggregateRepositoryProvider {
    /**
     * Get the aggregateRepository for the given aggregate type.
     */
    IAggregateRepositoryProxy getRepository(Class<? extends IAggregateRoot> aggregateRootType);
}

File: enode/src/main/java/org/enodeframework/domain/IAggregateRepositoryProvider.kt
package org.enodeframework.domain

/**
 * Represents a provider to provide the aggregate repository.
 */
interface IAggregateRepositoryProvider {
    /**
     * Get the aggregateRepository for the given aggregate type.
     */
    fun getRepository(aggregateRootType: Class<out IAggregateRoot>): IAggregateRepositoryProxy?
}